### PR TITLE
Calendario regional - aparece a veces el empty state aunque ya hay diccionarios

### PR DESCRIPTION
**Pull request by fermarinsanchez**
fix(academic-calendar): Create hook with React Query on Regional Calendar view to prevent accidentally Empty State and fix some issues.

### CODE REVIEW

File: plugins/leemons-plugin-academic-calendar/frontend/src/hooks/useListRegionalConfigs.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { useQuery } from '@tanstack/react-query';
+import { listRegionalConfigs } from '../request/regional-config';
+
+export const useListRegionalConfigs = (center, options = {}) =>
+  useQuery({
+    queryKey: ['listRegionalConfigs', center],
+    queryFn: () => listRegionalConfigs(center),
+    ...options,
+  });
```
Comment (MIGUELez11) [inline plugins/leemons-plugin-academic-calendar/frontend/src/hooks/useListRegionalConfigs.js:9]: React query key does not match the defined standard

File: plugins/leemons-plugin-academic-calendar/frontend/src/pages/private/regional/index.js [hunk 2]
```diff
@@ -120,10 +119,14 @@ export default function RegionalCalendars() {
...
+    store?.center?.id,
+    { enabled: !!store?.center?.id }
+  );
+
   async function loadRegionalConfigs() {
-    const { regionalConfigs } = await listRegionalConfigsRequest(store?.center?.id);
-    store.regionalConfigs = regionalConfigs;
-    setRegConfigs(regionalConfigs);
+    refetchConfigs();
+    store.regionalConfigs = regionalConfigs?.regionalConfigs;
```
Comment (MIGUELez11) [inline plugins/leemons-plugin-academic-calendar/frontend/src/pages/private/regional/index.js:129]: Does not work as expected, refetch function will trigger a refetch effectively, but it won't update regionalConfigs var inside of this context.

In other words, store.regionalConfigs will potentialy have a missmatching version of the regionalConfigs.

File: plugins/leemons-plugin-academic-calendar/frontend/src/pages/private/regional/index.js [hunk 3]
```diff
@@ -166,13 +169,20 @@ export default function RegionalCalendars() {
...
+      };
       try {
         store.saving = true;
         render();
 
         await saveRegionalConfig({
-          ...data,
-          center: store.center.id,
+          ...dataToSave,
         });
```
Comment (MIGUELez11) [inline plugins/leemons-plugin-academic-calendar/frontend/src/pages/private/regional/index.js:186]: If you are creating a new object to save, then you don't need to use object destructuring, as the object mutation won't affect you

**Review by MIGUELez11 — CHANGES_REQUESTED** [review]

File: plugins/leemons-plugin-academic-calendar/frontend/src/hooks/useListRegionalConfigs.js (hunk 1 above)
Comment (fermarinsanchez) [reply plugins/leemons-plugin-academic-calendar/frontend/src/hooks/useListRegionalConfigs.js:9]: implement queryKey standard

File: plugins/leemons-plugin-academic-calendar/frontend/src/pages/private/regional/index.js (hunk 2 above)
Comment (fermarinsanchez) [reply plugins/leemons-plugin-academic-calendar/frontend/src/pages/private/regional/index.js:129]: I've made a refactor on entire archive. Add React Query hooks to all queries and mutations. Remove store and renders.

File: plugins/leemons-plugin-academic-calendar/frontend/src/pages/private/regional/index.js (hunk 3 above)
Comment (fermarinsanchez) [reply plugins/leemons-plugin-academic-calendar/frontend/src/pages/private/regional/index.js:186]: I create a new hook called "useSaveRegionalConfig" and pass "dataToSave" directly. No destructuring object.

File: plugins/leemons-plugin-academic-calendar/frontend/src/hooks/keys/listRegionalConfigs.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+export const allListRegionalConfigsKeys = [
+  {
+    plugin: 'plugin.academic-calendar',
+    scope: 'list-regional-configs',
```
Comment (MIGUELez11) [inline plugins/leemons-plugin-academic-calendar/frontend/src/hooks/keys/listRegionalConfigs.js:4]:
```suggestion
    action: 'list',
    scope: 'regional-configs',
```

File: plugins/leemons-plugin-academic-calendar/frontend/src/pages/private/regional/index.js [hunk 5]
```diff
@@ -106,65 +105,53 @@ export default function RegionalCalendars() {
...
-      await loadRegionalConfigs();
-      render();
-    } catch (error) {
-      console.error('Error deleting regional config:', error);
-      addErrorAlert(getErrorMessage(error));
-    }
+  function handleOnDeleteRegionalCalendar(configId) {
+    deleteRegionalConfigs(configId, center?.id, {
+      onSuccess: () => {
+        addSuccessAlert('Config deleted successfully');
```
Comment (MIGUELez11) [inline plugins/leemons-plugin-academic-calendar/frontend/src/pages/private/regional/index.js:130]: Copy not localized

**Review by MIGUELez11 — CHANGES_REQUESTED** [review]

File: plugins/leemons-plugin-academic-calendar/frontend/src/hooks/keys/listRegionalConfigs.js (hunk 4 above)
Comment (fermarinsanchez) [reply plugins/leemons-plugin-academic-calendar/frontend/src/hooks/keys/listRegionalConfigs.js:4]: Done.

File: plugins/leemons-plugin-academic-calendar/frontend/src/pages/private/regional/index.js (hunk 5 above)
Comment (fermarinsanchez) [reply plugins/leemons-plugin-academic-calendar/frontend/src/pages/private/regional/index.js:130]: localization added, and create a new localization for error alert.

File: plugins/leemons-plugin-academic-calendar/frontend/src/hooks/keys/listRegionalConfigs.js [hunk 6]
```diff
@@ -1,7 +1,7 @@
 export const allListRegionalConfigsKeys = [
   {
     plugin: 'plugin.academic-calendar',
-    scope: 'list-regional-configs',
+    scope: 'list',
```
Comment (MIGUELez11) [inline plugins/leemons-plugin-academic-calendar/frontend/src/hooks/keys/listRegionalConfigs.js:4]:
```suggestion
    action: 'list',
    scope: 'regional-config'
```

**Review by MIGUELez11 — CHANGES_REQUESTED** [review]

File: plugins/leemons-plugin-academic-calendar/frontend/src/hooks/keys/listRegionalConfigs.js (hunk 4 above)
Comment (MIGUELez11) [reply plugins/leemons-plugin-academic-calendar/frontend/src/hooks/keys/listRegionalConfigs.js]: Not how I indicated :D, I fix it, no problem 😉

File: plugins/leemons-plugin-academic-calendar/frontend/src/hooks/keys/listRegionalConfigs.js (hunk 6 above)
Comment (MIGUELez11) [reply plugins/leemons-plugin-academic-calendar/frontend/src/hooks/keys/listRegionalConfigs.js:4]: Well, I'm technically wrong, in the all the action should not exists, but in the getList

**Review by MIGUELez11 — APPROVED** [review]

**Pull request merged**